### PR TITLE
chore(cli): rename build options to ephemeral parameters in cli

### PR DESCRIPTION
**Pull request by defelmnq**
This PR aims to rename `build-option` to `ephemeral-parameters` based on #10488 conversation.

`build-option` has been renamed `ephemeral-parameter` and can be used to define a value for an ephemeral parameter in the template.

`build-options` has been renamed `prompt-ephemeral-parameters` and can be used to prompt the user to put values for the ephemeral parameters in the template.

### CODE REVIEW

File: docs/admin/templates/extending-templates/parameters.md
Comment (dannykopping) [inline docs/admin/templates/extending-templates/parameters.md:207]: @EdwardAngert would you mind weighing in on this?

Comment (defelmnq) [reply docs/admin/templates/extending-templates/parameters.md:207]: Thanks, definitely up for some help here.

The main point @EdwardAngert is that I'd like to remove anything related to build_options. For the rest, really keen for some help but nothing has changed per itself in the logic.

File: cli/parameter.go
Comment (mafredri) [inline cli/parameter.go:55]:
```suggestion
			Description: `Set the value of ephemeral parameters defined in the template. The format is "name=value".`,
```
I felt this was a bit vague, as if new parameters unrelated to the template could be defined. This is my take but can likely be improved further.

File: cli/parameter.go
Comment (mafredri) [inline cli/parameter.go:61]:
```suggestion
			Description: "Prompt to set values of ephemeral parameters defined in the template. If a value has been set via --ephemeral-parameter, it will not be prompted for.",
```
Not sure if the latter is true, but as a user I would like to understand the relationship between these two when reading the help.

**Review by mafredri — COMMENTED** [review]
Left a few suggestions, but otherwise looks good 👍

File: cli/parameter.go
Comment (defelmnq) [reply cli/parameter.go:61]: Hey @mafredri from what I've seen in the code - I confirm the behavior you described and love your suggestion. Thanks 🙏

**Comment by defelmnq** [comment]
Thanks @mafredri - your suggestions improved the usage of it.

Just waiting for some feedback in the documentation, but all good for me otherwise.

**Review by mtojek — COMMENTED** [review]
@defelmnq Looks like CI is complaining about some missing changes.

**Review by EdwardAngert — APPROVED** [review]

**Comment by defelmnq** [comment]
Just fixed the CI here - logic and test side should be good - just waiting for a last review from @EdwardAngert as now the CI is happy and it will simplify changes.

File: cli/restart_test.go
Comment (dannykopping) [inline cli/restart_test.go:130]: We might want one test to validate that `--build-option` also still works.

**Review by dannykopping — APPROVED** [review]
LGTM, let's add a test to ensure the deprecated field still works.
Doesn't need to be exhaustive like the current tests, just one will suffice.

**Review by EdwardAngert — APPROVED** [review]

File: cli/restart_test.go
Comment (defelmnq) [reply cli/restart_test.go]: Just added back a test for both `--build-option` and `--build-options`. 🙌

**Review by dannykopping — APPROVED** [review]
LGTM! Thanks a lot @defelmnq

**Pull request merged**
